go vet fails on GOARCH 386

```
$ GOARCH=386 go vet .
# pack.ag/amqp
./client.go:197:8: constant 4294967296 overflows int

```

Simple fix should be

```
@@ -189,12 +189,12 @@ func SessionOutgoingWindow(window uint32) SessionOption {
// n must be in the range 1 to 4294967296.
//
// Default: 4294967296.
-func SessionMaxLinks(n int) SessionOption {
+func SessionMaxLinks(n uint32) SessionOption {
	return func(s *Session) error {
		if n < 1 {
			return errorNew("max sessions cannot be less than 1")
		}
-		if n > 4294967296 {
+		if n > math.MaxUint32 {
			return errorNew("max sessions cannot be greater than 4294967296")
		}
s.handleMax = uint32(n - 1)
```

--- a/Makefile
+++ b/Makefile
@@ -20,7 +20,12 @@ test:
 	go test -tags gofuzz -v -race ./...
 
 integration:
-	go test -tags "integration pkgerrors" -count=1 -v -race ./...
+	go test -tags "integration pkgerrors" -count=1 -v -race .
+
+test386:
+	TEST_CORPUS=1 go test -tags "gofuzz" -count=1 -v .
+
+ci: test386 coverage
 
 coverage:
 	TEST_CORPUS=1 go test -tags "integration gofuzz" -cover -coverprofile=cover.out -v

--- a/client.go
+++ b/client.go
@@ -194,7 +194,7 @@ func SessionMaxLinks(n int) SessionOption {
 		if n < 1 {
 			return errorNew("max sessions cannot be less than 1")
 		}
-		if n > 4294967296 {
+		if int64(n) > 4294967296 {
 			return errorNew("max sessions cannot be greater than 4294967296")
 		}
 		s.handleMax = uint32(n - 1)

